zebrad/src/components/inbound.rs

```diff
@@ -258,38 +277,46 @@ impl Service<zn::Request> for Inbound {
     /// and will cause callers to disconnect from the remote peer.
     #[instrument(name = "inbound", skip(self, req))]
     fn call(&mut self, req: zn::Request) -> Self::Future {
+        let (address_book, block_downloads, mempool, state) = match &mut self.setup {
+            Setup::Initialized {
+                address_book,
+                block_downloads,
+                mempool,
+                state,
+            } => (address_book, block_downloads, mempool, state),
+            _ => {
+                debug!("ignoring request from remote peer during setup");
+                return async { Ok(zn::Response::Nil) }.boxed();
+            }
+        };
+
         match req {
             zn::Request::Peers => {
-                if let Setup::Initialized { address_book, .. } = &self.network_setup {
-                    // # Security
```

Comment:
There is a lot of code movement in this file, so it might be easier to review with:
```sh
git diff --color-moved
```